### PR TITLE
Update dependencies, remove unused dependencies

### PR DESCRIPTION
**Pull request by pitaj**
- use @nodebb/mubsub to support mongodb@3
- remove unused async, mongodb-uri
- add test-running information to README

Please release a new major version with these fixes. Thanks!

### CODE REVIEW

**Comment by barisusakli** [comment]
@lklepner any chance of merging this in?

**Comment by lklepner** [comment]
Hi Bans, Sorry for the delay in responding.  I have not been a user of the
module for several years now. Any chance you’d be interested in taking
ownership of it?

**Comment by barisusakli** [comment]
Sure, let me know if you need anything from me. We would probably publish it as @nodebb/socket.io-adapter-mongo

**Comment by lklepner** [comment]
Awesome, moving the publishing origin makes sense. I just sent you an email
to get your npmjs username.

**Comment by lklepner** [comment]
@pitaj Sorry for the delay in responding.  I have not been a user of the
module for several years now. Any chance you’d be interested in taking
ownership of it? If so please send me your mom username and I’ll initiate the transfer on my end.

**Comment by gsRatti1** [comment]
Hello @pitaj, any chance of merging this in soon ?

Right now what I have done is that I am passing my own mubsub client (nodebb/mubsub), so that socket io mongo adpater does not rely on the old mubsub library
Something like this
```

import * as mubsub from '@nodebb/mubsub';
const mubsubClient = mubsub(<conn-string>);
socketServer.adapter(SocketMongoAdaptor(<conn-string>, { client: mubsubClient }));
```

But I see that you have bumped up the version of certain other dependencies too in this PR.
Do you think doing what I have done could cause issues ?
Actually I am seeing some flakiness in ws communication but I am not entirely sure if it is because of this.

File: package.json [hunk 1]
```diff
@@ -3,22 +3,20 @@
...
   },
   "dependencies": {
-    "debug": "2.6.9",
-    "mongodb-uri": "^0.9.7",
+    "debug": "4.1.1",
     "msgpack-js": "0.3.0",
-    "mubsub": "^1.4.0",
-    "socket.io-adapter": "^1.1.1",
+    "@nodebb/mubsub": "^1.6.0",
+    "socket.io-adapter": "^1.1.2",
```
Comment (gsRatti1) [inline package.json:12]: What if we dont bump up the version of socket io adapter ? Will it cause issues with nodebb/mubsub ?

File: package.json [hunk 2]
```diff
@@ -3,22 +3,20 @@
...
   },
   "devDependencies": {
-    "async": "0.2.10",
     "expect.js": "0.3.1",
-    "mocha": "1.18.0",
-    "socket.io": "^2.0.3",
-    "socket.io-client": "^2.0.3"
+    "mocha": "7.1.1",
+    "socket.io": "^2.3.0",
+    "socket.io-client": "^2.3.0"
```
Comment (gsRatti1) [inline package.json:19]: Same question here

**Comment by pitaj** [comment]
NodeBB doesn't use this adapter anymore and I doubt it's compatible with newer versions of socket.io

I believe @nodebb/socket.io-adapter-mongo is still published if you want to try that.

**Comment by gsRatti1** [comment]
Thanks for the suggestion @pitaj .

Do you perhaps know how we can set the recreate option to true while creating the mubsub channel.
The adapter library internally creates this channel, and does not pass this option



When I try to add that option via a patch, it gives me this error : MongoError: BSON field 'create.recreate' is an unknown field.
The mubsub library says it is true by default but that does look like it

Basically I am trying this :

const mubsubClient = mubsub(<conn-string>);
const channel= mubsubClient.channel ('test-channel', { recreate: true})

**Comment by pitaj** [comment]
This really isn't the place to be asking questions. Please direct them to the NodeBB adapter issues page.

You can pass your own mubsub object in as `opts.client`, otherwise the adapter creates it's own client. https://github.com/NodeBB/socket.io-adapter-mongo/blob/818a76e4c87b6b94e87111bd86b4d492c6531584/index.js#L36
